TRB_label_assist/models.py

```diff
@@ -367,7 +382,7 @@ def _match(self, trip, bin, loc_type):
             copied from the Similarity class on the e-mission-server. 
         """
         for t_idx in bin:
-            trip_in_bin = self.train_df.iloc[t_idx]
+            trip_in_bin = self.train_df.iloc[int(t_idx)]
```

Comment:
the bins in `bin` are stored as list of strings, for e.g. ['1','2','3'....]

Reply:
so how did it work before?

Reply:
the `tour_model` generated bins in the form [1,2,3 ...] . 
 previously , this is how we got bins from the model:
`labels = sim_model.data_df[loc_type + '_bin'].to_list()`.    ( Before changes Line 112 in [clustering.py ](https://github.com/e-mission/e-mission-eval-private-data/pull/37/files) )


The 'trip_model' saves it as mentioned in my previous comment above. 
this is how we got bins from the model now :
 'labels = [int(l) for l in sim_model.tripLabels]'.  ( After changes Line 140 in [clustering.py ](https://github.com/e-mission/e-mission-eval-private-data/pull/37/files) )

Reply:
so the difference here is that we are using strings instead of ints?

Reply:
the way GreedySimilarityBinning was implemented, these are stored as strings. But converted to ints when required.

Reply:
I am not going to hold up the merge for this, but I don't understand why we can't just convert from strings to ints one time right after return instead of converting for each usage.

Reply:
Sure, we can. These bins are stored in the `self.bins` variable . We access these bins in the predict function. 

So, we can change the keys from string to int right before we begin our prediction.